components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5777,12 +5778,8 @@ public boolean isRecommendationEnabled(String tenantDomain) {
                 return true;
             } else {
                 try {
-                    org.json.JSONObject tenantConfig = null;
-                    Cache tenantConfigCache = APIUtil.getCache(
-                            APIConstants.API_MANAGER_CACHE_MANAGER,
-                            APIConstants.TENANT_CONFIG_CACHE_NAME,
-                            APIConstants.TENANT_CONFIG_CACHE_MODIFIED_EXPIRY,
-                            APIConstants.TENANT_CONFIG_CACHE_ACCESS_EXPIRY);
+                    org.json.JSONObject tenantConfig;
+                    Cache tenantConfigCache = CacheProvider.getTenantConfigCache();
```

Comment:
make new method to abstract apiutil.getTenantConfig

Reply:
Fixed